### PR TITLE
#6 Updating trainee profile data

### LINKED ISSUE #6: Update trainee profile
# Description

- As a trainee, I want to update my profile information.
- I should be able to view updated info
- Appropriate errors should be displayed

### PR DESCRIPTION
**Pull request by munezeromicha**
### What does this PR do?
This PR allows users to update their profile information, including:

- First Name
- Last Name
- Telephone
- GitHub Username
- Address
- City
- Country
- Biography

It integrates the updated form with the backend using GraphQL and Apollo Client, ensuring that users' profile data is fetched and updated correctly.

### Update Profile Form:

- Added form fields for First Name, Last Name, Telephone, GitHub Username, Address, City, Country, and Biography.
- Pre-populated the fields with existing user data using GET_PROFILE GraphQL query.
- Integrated form submission using the UPDATE_PROFILE mutation to update the user’s information.

### UI/UX Enhancements:

- Styled the form using Tailwind CSS for responsive design in both dark and light modes.
- Included form validation with Formik and Yup to ensure valid input before submission.

### Data Fetching & State Management:

- Utilized Apollo Client to manage fetching and updating profile data.
- Ensured the form automatically re-populates with updated data after submission.

### Fetch the branch:

`git checkout ft-update-trainee-profile`

**Run the app and navigate to the profile editing page.**

### Update profile fields:

- Change the values for First Name, Last Name, Telephone, GitHub Username, Address, City, Country, or Biography.
- Submit the form.

### Verify:

- Ensure the updated profile information appears on the profile page after submission.
- Confirm that toast notifications appear for success and error handling.

### Screenshots:

<div style="display: flex; flex-wrap: wrap;">
    Screenshot 1
    Screenshot 2
</div>

### CODE REVIEW

File: app/dashboard/EditProfile/EditProfile.tsx [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+      </View>
+
+      {tab === 0 && (
+        <View className={`bg-[#272728] px-4 pt-10 pb-6 rounded-lg ${bgColor} shadow-lg`}>
+          <TouchableOpacity
+            onPress={() => router.push('/dashboard/trainee/Profile')}
+            className="flex-row items-center mb-4 "
+          >
+            <Text className="text-white text-lg bg-action-500 p-2 rounded-lg">
+              ← Back to Profile
```
Comment (niyibi250) [inline app/dashboard/EditProfile/EditProfile.tsx]: what is this is this icon , please  icon not this

File: app/dashboard/EditProfile/EditProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+                value={formik.values.lastName}
+              />
+            </View>
+            <Text className="text-error-500 my-1">{formik.errors.lastName}</Text>
+
+            <Text className={`${textColor} pb-2`}>Telephone</Text>
+            <View
+              className={`flex-row items-center ${bgColor} rounded-lg shadow border-2 border-[#D2D2D2]`}
+            >
+              <TextInput
```
Comment (niyibi250) [inline app/dashboard/EditProfile/EditProfile.tsx:270]: when i update the phone number there not being updated ???

File: app/dashboard/EditProfile/EditProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+                keyboardType="numeric"
+                maxLength={13}
+              />
+            </View>
+
+            <Text className={`${textColor} pb-2 pt-4`}>GitHub Username</Text>
+            <View
+              className={`flex-row items-center ${bgColor} rounded-lg shadow border-2 border-[#D2D2D2]`}
+            >
+              <TextInput
```
Comment (niyibi250) [inline app/dashboard/EditProfile/EditProfile.tsx:286]: githab  user name too not being updated

**Review by niyibi250 — CHANGES_REQUESTED** [review]
1. use reuseble component , to make file short like text input  create component then use it.
2.on edit profile header on right side the comera icon is missing look at it: 
 https://www.figma.com/design/Y6yBiYmYqJvlKhf37eOqoo/Igor-steppers?node-id=1149-5193&t=xyrMOXOp0EuNIidd-4

File: app/dashboard/EditProfile/EditProfile.tsx [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+
+  const { data, loading, error } = useQuery(GET_PROFILE, {
+    context: {
+      headers: {
+        Authorization: `Bearer ${userToken}`,
+      },
+    },
+    skip: !userToken,
+  });
+  console.log(data);
```
Comment (pacifiquemboni) [inline app/dashboard/EditProfile/EditProfile.tsx]: remove this console

File: app/dashboard/EditProfile/EditProfile.tsx (hunk 1 above)
Comment (munezeromicha) [reply app/dashboard/EditProfile/EditProfile.tsx]: **Done**

File: app/dashboard/EditProfile/EditProfile.tsx (hunk 2 above)
Comment (munezeromicha) [reply app/dashboard/EditProfile/EditProfile.tsx:270]: **Done**

File: app/dashboard/EditProfile/EditProfile.tsx (hunk 4 above)
Comment (munezeromicha) [reply app/dashboard/EditProfile/EditProfile.tsx]: **Done**

**Review by Mbabazi12 — APPROVED** [review]

**Review by pacifiquemboni — APPROVED** [review]

**Review by MugemaneBertin2001 — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]

**Pull request merged**